crates/ruff_python_semantic/src/model.rs

```diff
@@ -766,13 +766,25 @@ impl<'a> SemanticModel<'a> {
             }
             BindingKind::ClassDefinition(_) | BindingKind::FunctionDefinition(_) => {
                 let value_name = UnqualifiedName::from_expr(value)?;
-                let resolved: QualifiedName = self
-                    .module_path?
-                    .iter()
-                    .map(String::as_str)
-                    .chain(value_name.segments().iter().copied())
-                    .collect();
-                Some(resolved)
+
+                // If we have a fully-qualified path for the module, use it.
+                if let Some(path) = self.module.path() {
```

Comment:
I consider changing `Module` to return `&[script_name]` in the script case, but the path is typically used for resolving relative imports, and I'm not sure it makes as much sense for us to resolve relative imports relative to the script name given that it's definitionally a standalone script.